### ISSUE TITLE
nodejs build runs after test

### ISSUE THREAD

**Issue by chipit24** [travis-ci/travis-ci#7854]
This is my `.travis.yml` file:
```
language: node_js
node_js:
  - "7"
  - "6"
script:
  - npm run build 
  - npm test
```

However, travis always errors out with `The command "eval yarn " failed`, and I see that it runs `npm test && npm run build`. I need it to run `npm run build` before `npm test`, how can I get it to do this?

I've tried a few variations of the config but it always fails with the same error, and I always see it running `npm test && npm run build`.

**Comment by BanzaiMan** [comment]
Do you have a build log URL that shows the problem you are describing here?

**Comment by chipit24** [comment]
Here's the link to the build: https://travis-ci.org/chipit24/fade-props/jobs/239442866
In my last attempt, I modified my config to `npm run build && npm test`, but the errors are still the same.

```
{
  "language": "node_js",
  "node_js": "7",
  "script": [
    "npm run build && npm test"
  ],
  "group": "stable",
  "dist": "precise",
  "os": "linux"
}
```

**Comment by BanzaiMan** [comment]
Your build is not getting to the `script` phase, but failing in the `install` phase, which runs `yarn`, based on the repository configuration. https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Travis-CI-supports-yarn

If you don't want to run this command, skip it.

```yaml
language: node_js
install: skip
⋮
```

**Comment by chipit24** [comment]
So the problem was that `npm install` (or just `yarn`) also runs `prepublish`, which I had set to `npm test && npm run build`. I just removed `prepublish` and replaced it with `"prepublishOnly": "npm run build && npm test"`. I then updated my `.travis.yml` to the below, and everything works now 🎉

```
language: node_js
node_js:
  - "7"
  - "6"
before_script:
  - npm run build
```

I was unaware of the above behaviour with npm and originally thought it was an issue with travis-ci.
